src/main/kotlin/no/nav/syfo/veileder/VeilederApi.kt

```diff
@@ -1,31 +1,87 @@
 package no.nav.syfo.veileder
 
+import io.ktor.http.ContentType
+import io.ktor.http.HttpStatusCode
+import io.ktor.server.application.ApplicationCall
 import io.ktor.server.application.call
-import io.ktor.server.application.install
 import io.ktor.server.auth.authenticate
-import io.ktor.server.response.respondText
+import io.ktor.server.response.respond
+import io.ktor.server.response.respondBytes
 import io.ktor.server.routing.Routing
 import io.ktor.server.routing.get
 import io.ktor.server.routing.route
-import no.nav.syfo.application.api.VeilederTilgangskontrollPlugin
 import no.nav.syfo.application.api.auth.JwtIssuerType
+import no.nav.syfo.application.database.DatabaseInterface
+import no.nav.syfo.application.exception.ForbiddenAccessVeilederException
 import no.nav.syfo.client.veiledertilgang.VeilederTilgangskontrollClient
+import no.nav.syfo.domain.PersonIdent
+import no.nav.syfo.util.NAV_PERSONIDENT_HEADER
+import no.nav.syfo.util.getBearerHeader
+import no.nav.syfo.util.getCallId
+import no.nav.syfo.util.getPersonIdent
+import no.nav.syfo.veileder.database.getOppfolgingsplanPdf
+import no.nav.syfo.veileder.database.getOppfolgingsplanerForVeileder
+import java.util.*
 
-const val VEILEDER_LPS_METADATA_PATH = "/api/internad/v1/oppfolgingsplan/lps"
+const val VEILEDER_LPS_BASE_PATH = "/api/internad/v1/oppfolgingsplan/lps"
+const val VEILEDER_LPS_UUID_PARAM = "uuid"
 private const val API_ACTION = "access lps plan for person"
 
 fun Routing.registerVeilederApi(
     veilederTilgangskontrollClient: VeilederTilgangskontrollClient,
+    database: DatabaseInterface,
 ) {
-    route(VEILEDER_LPS_METADATA_PATH) {
+    route(VEILEDER_LPS_BASE_PATH) {
         authenticate(JwtIssuerType.INTERNAL_AZUREAD.name) {
-            install(VeilederTilgangskontrollPlugin) {
-                this.action = API_ACTION
-                this.veilederTilgangskontrollClient = veilederTilgangskontrollClient
-            }
             get {
-                call.respondText("OK")
+                val callId = call.getCallId()
+                val personIdent = call.personIdent()
+                val token = call.bearerHeader()
+
+                val hasAccess = veilederTilgangskontrollClient.hasAccess(
+                    callId = callId,
+                    personIdent = personIdent,
+                    token = token,
+                )
+
+                if (!hasAccess) {
+                    throw ForbiddenAccessVeilederException(
+                        action = API_ACTION,
+                    )
+                }
+
+                val plansSharedWithNAV = database
+                    .getOppfolgingsplanerForVeileder(personIdent)
+                call.respond(plansSharedWithNAV)
+            }
+
+            get("/{$VEILEDER_LPS_UUID_PARAM}") {
+                val planUUID = call.uuid()
+                val plan = database.getOppfolgingsplanPdf(planUUID)
+
+                val personIdent = plan?.first ?: throw IllegalArgumentException(
```

Comment:
Kan evnt vurdere require i stedet for throw. Gjør jo det samme, men har tidligere syntes det kanskje blir hakket ryddigere